### ISSUE TITLE
fish: environment.shellAliases are not available in non-login shell

### ISSUE THREAD

**Issue by nagisa** [NixOS/nixpkgs#25800]
## Issue description

Fish usually has `environment.shellAliases` available, but the aliases are now missing, when the shell is not a login shell (presumably because of https://github.com/NixOS/nixpkgs/pull/24314).

### Steps to reproduce

Have

```
  environment.shellAliases = {
      lsa = "${pkgs.coreutils}/bin/ls -lahFT0 --group-directories-first";
  };
```

Then launch `fish` as a not-login shell. Try to use `lsa`.

## Technical details

* System: 17.09pre107265.0afb6d789c (Hummingbird)

**Comment by therealpxc** [comment]
Agh, what a mess. Thanks for raising this issue.

**Comment by therealpxc** [comment]
I think this issue is a little subtler than one would think: aliases are sourced, but only when the parent shell has not already sourced the interactive configuration.

This is out of line with the behavior of the other shells on NixOS. I have a fix which I've tested with your example here. I will post a link to the relevant PR momentarily (after I put it up).

**Comment by therealpxc** [comment]
So apparently outgoing SSH is blocked from the WiFi on this cruise ship, so I can't push commits with the remotes I have set up, and my sister is nagging me to get off the computer so we can go out. If you need a quick fix on your own machine (I will be back later today, while I still have internet access, to submit the new PR), just change the `set -gx <...>` lines in the NixOS module for fish (<nixpkgs>/nixos/modules/programs/fish.nix) to  `set -g <...>` and your aliases will get sourced every time the shell starts, so changes will be reflected without a reboot or manually unsetting the relevant variable before starting a new shell (`set -e __fish_nixos_interactive_config_sourced`).

(The latter will do the trick if you want a quick fix without any changes to your Nixpkgs tree.)

**Comment by therealpxc** [comment]
there's a fix in this commit: https://github.com/NixOS/nixpkgs/pull/25835/commits/183082c1f872f3c6fe2aab10f176d623c8178f80

**Comment by therealpxc** [comment]
Thanks!
